addons/addon-base-raas/packages/base-raas-services/lib/migration/migration-service.js

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const Service = require('@aws-ee/base-services-container/lib/service');
+const { isAdmin } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+
+const settingKeys = { studiesTableName: 'dbStudies', categoryIndexName: 'dbStudiesCategoryIndex' };
+
+class MigrationService extends Service {
+  constructor() {
+    super();
+    this.dependency(['studyOperationService', 'dbService', 'userService', 'studyService']);
+  }
+
+  async init() {
+    await super.init();
+    const dbService = await this.service('dbService');
+
+    const studiesTable = this.settings.get(settingKeys.studiesTableName);
+    this._getter = () => dbService.helper.getter().table(studiesTable);
+    this._updater = () => dbService.helper.updater().table(studiesTable);
+    this._query = () => dbService.helper.query().table(studiesTable);
+    this._deleter = () => dbService.helper.deleter().table(studiesTable);
+    this._scanner = () => dbService.helper.scanner().table(studiesTable);
+
+    this.categoryIndex = this.settings.get(settingKeys.categoryIndexName);
+  }
+
+  async migrateMyStudiesPermissions(requestContext, migrationMappings) {
+    if (!isAdmin(requestContext)) {
+      throw this.boom.forbidden("You don't have permission to migrate My Studies", true);
+    }
+    const studyOperationService = await this.service('studyOperationService');
+    const userService = await this.service('userService');
+    const studyService = await this.service('studyService');
+
+    const migrationResults = [];
+    await Promise.all(
+      migrationMappings.map(async item => {
+        const endUser = item.uid;
+        const studyId = item.studyId;
+        // console.log(result1);
+        if (await userService.isInternalAuthUser(endUser)) {
+          throw this.boom.forbidden('You cannot migrate My Studies to internal user', true);
+        }
+
+        // check that end user is not already owner
+        if (await studyService.isStudyAdmin(requestContext, studyId, endUser)) {
+          throw this.boom.forbidden('You cannot migrate My Studies to current owner', true);
+        }
+
+        const result = await studyOperationService.updatePermissions(requestContext, studyId, {
+          usersToAdd: [{ uid: endUser, permissionLevel: 'admin' }],
+          usersToRemove: [{ uid: '*', permissionLevel: 'admin' }],
```

Comment:
I'm not familiar with study permissions. What does a `uuid` of `*` mean? In what cases does a study have an owner of `*`?

Reply:
This is a new convention. The `*` means to remove all users like a wildcard. In the case of the My Studies, this will only be one user. But we don't necessarily know the current user so we need to put something. The logic in the methods file later adds whoever was the old admin of the My Study to be the value in place of `*`. 

TLDR: `*` is never the owner--just a wildcard placeholder.